<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Откройте-таблицу-и-изучите-общую-информацию-о-данных" data-toc-modified-id="Откройте-таблицу-и-изучите-общую-информацию-о-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Откройте таблицу и изучите общую информацию о данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Удаление-пропусков" data-toc-modified-id="Удаление-пропусков-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Удаление пропусков</a></span></li><li><span><a href="#Обработка-аномальных-значений" data-toc-modified-id="Обработка-аномальных-значений-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Обработка аномальных значений</a></span></li><li><span><a href="#Удаление-пропусков-(продолжение)" data-toc-modified-id="Удаление-пропусков-(продолжение)-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Удаление пропусков (продолжение)</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-2.6"><span class="toc-item-num">2.6&nbsp;&nbsp;</span>Категоризация данных</a></span></li><li><span><a href="#Шаг-3.-Исследуйте-данные-и-ответьте-на-вопросы" data-toc-modified-id="Шаг-3.-Исследуйте-данные-и-ответьте-на-вопросы-2.7"><span class="toc-item-num">2.7&nbsp;&nbsp;</span>Шаг 3. Исследуйте данные и ответьте на вопросы</a></span><ul class="toc-item"><li><span><a href="#3.1-Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?" data-toc-modified-id="3.1-Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?-2.7.1"><span class="toc-item-num">2.7.1&nbsp;&nbsp;</span>3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?</a></span></li><li><span><a href="#3.2-Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?" data-toc-modified-id="3.2-Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?-2.7.2"><span class="toc-item-num">2.7.2&nbsp;&nbsp;</span>3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?</a></span></li><li><span><a href="#3.3-Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?" data-toc-modified-id="3.3-Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?-2.7.3"><span class="toc-item-num">2.7.3&nbsp;&nbsp;</span>3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?</a></span></li><li><span><a href="#3.4-Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?" data-toc-modified-id="3.4-Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?-2.7.4"><span class="toc-item-num">2.7.4&nbsp;&nbsp;</span>3.4 Как разные цели кредита влияют на его возврат в срок?</a></span></li><li><span><a href="#3.5-Приведите-возможные-причины-появления-пропусков-в-исходных-данных." data-toc-modified-id="3.5-Приведите-возможные-причины-появления-пропусков-в-исходных-данных.-2.7.5"><span class="toc-item-num">2.7.5&nbsp;&nbsp;</span>3.5 Приведите возможные причины появления пропусков в исходных данных.</a></span></li><li><span><a href="#3.6-Объясните,-почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных." data-toc-modified-id="3.6-Объясните,-почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных.-2.7.6"><span class="toc-item-num">2.7.6&nbsp;&nbsp;</span>3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.</a></span></li></ul></li><li><span><a href="#Шаг-4:-общий-вывод." data-toc-modified-id="Шаг-4:-общий-вывод.-2.8"><span class="toc-item-num">2.8&nbsp;&nbsp;</span>Шаг 4: общий вывод.</a></span></li></ul></li></ul></div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
##доля заемщиков с прошлыми задолжностями в зависимости от числа детей
print(data.groupby("children")["debt"].sum() / data.groupby("children")["debt"].count(), "\n")

print(data.groupby("children")["children"].count())

children
0    0.075438
1    0.092346
2    0.094542
3    0.081818
4    0.097561
5    0.000000
Name: debt, dtype: float64 

children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: children, dtype: int64


Наблюдения:
* Доли заемщиков с прошлыми задолжностями, у которых 4 и более детей неприменимы ввиду недостаточного количества имеющихся данных (менее 0.25% датасета). В дальнейшем анализе эти группы участовать не будут.
* Вероятно, аналогичная ситуация наблюдается и с заемщиками с 3 детьми. Их доля в датасете составляет менее 1.7%. Получившаяся доля для этой группы может быть ошибочной.
* Доли заемщиков с прошлыми задолжностями с количеством детей 1 или 2, в целом, сравнимы.
* Доля заемщиков с прошлыми задолжностями без детей значительно ниже, чем у заемщиков с детьми.

**Вывод:** 
Влияет не столько количество детей, сколько сам факт их наличия.<br> Доля заемщиков с прошлыми задолжностями без детей существенно ниже, чем оных с детьми. При этом между заемщиками с разным количеством детей (>0) существенной разницы не наблюдается. Исключением может быть только группа с 3 детьми, однако, есть сомнения в правдивости данных для этой группы ввиду их количества.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:
print(data.groupby("family_status")["debt"].sum() / data.groupby("family_status")["debt"].count(), "\n")

print(data.groupby("family_status")["family_status"].count())

family_status
Не женат / не замужем    0.097639
в разводе                0.070648
вдовец / вдова           0.066246
гражданский брак         0.093130
женат / замужем          0.075606
Name: debt, dtype: float64 

family_status
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
гражданский брак          4134
женат / замужем          12261
Name: family_status, dtype: int64


Наблюдения:
* В целом, распределение получилось ~биноминальным~ бимодальным:
* 1 группа - ~9,5% - холостые и в гражданском браке
* 2 группа - ~7.2% - женатые и разведенные
* Особняком стоят вдовы, среди них доля должников минимальная. Возможно, это связано с возрастом или наследством от супруга(и)?

**Вывод:** 
Зависимость между семейным положением и возвратом кредита в срок есть.<br> Заемщики, не состоявшие в браке или состоящие в гражданском браке существенно менее надежны, чем заемщики, состоящие в браке или состоявшие в браке. При этом наиболее надежными заемщиками являются вдовы.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [24]:
print(data.groupby("total_income_category")["debt"].sum() / data.groupby("total_income_category")["debt"].count(), "\n")

print(data.groupby("total_income_category")["total_income_category"].count())

total_income_category
A    0.080000
B    0.070602
C    0.084982
D    0.060172
E    0.090909
Name: debt, dtype: float64 

total_income_category
A       25
B     5014
C    15921
D      349
E       22
Name: total_income_category, dtype: int64


Наблюдения:
* Количество данных о заемщиках с категориями доходов А и Е недостаточно для анализа.
* Аналогичная ситуация с категорией D, к ее показателям стоит относиться с осторожностью.

**Вывод:** 
На данном датасете сделать однозначный вывод затруднительно.<br>
Если подразумевать, что доля должников в категории D неверна из-за количества имеющихся данных (занижена), то с увеличением дохода наблюдается монотонное уменьшение доли должников.<br>
Однако, если считать, что показатели кат. D верны, то получается, что монотонной зависимости нет.<br>
С уверенностью можно сказать одно: категория B в целом существенно более платежеспособна, чем категория C.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [25]:
print(data.groupby("purpose_category")["debt"].sum() / data.groupby("purpose_category")["debt"].count(), "\n")

print(data.groupby("purpose_category")["purpose_category"].count())

purpose_category
операции с автомобилем      0.093480
операции с недвижимостью    0.072551
получение образования       0.092528
проведение свадьбы          0.079118
Name: debt, dtype: float64 

purpose_category
операции с автомобилем       4279
операции с недвижимостью    10751
получение образования        3988
проведение свадьбы           2313
Name: purpose_category, dtype: int64


Наблюдения:<br>
Снова довольно бимодальное распределение с 2 группами:
* Образование и автомобили - ~9.2% долгов
* Свадьба и недвижимость - ~7.5% долгов

**Вывод:** зависимость между целью кредита и возвратом в срок есть. <br>
Доля кредитов, полученных с целью операций с авто или получением образования, по которым образовалась задолженность, существенно выше, чем доля кредитов, полученных для операций с недвижимостью или свадеб.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

In [26]:
try:
    dummy = pd.read_csv('/datasets/data.csv')
except:
    dummy = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')
    
print(dummy.loc[dummy["days_employed"].isna(), ["days_employed", "total_income"]].head(25))
del(dummy)

     days_employed  total_income
12             NaN           NaN
26             NaN           NaN
29             NaN           NaN
41             NaN           NaN
55             NaN           NaN
65             NaN           NaN
67             NaN           NaN
72             NaN           NaN
82             NaN           NaN
83             NaN           NaN
90             NaN           NaN
94             NaN           NaN
96             NaN           NaN
97             NaN           NaN
120            NaN           NaN
121            NaN           NaN
135            NaN           NaN
141            NaN           NaN
145            NaN           NaN
174            NaN           NaN
181            NaN           NaN
189            NaN           NaN
205            NaN           NaN
220            NaN           NaN
241            NaN           NaN


*Ответ:* Учитывая, что пропуски в столбцах days_employed и total_income синхронизированы (см. аутпут ячейки выше), то наиболее вероятным сценарием выглядит техническая ошибка при выгрузке данных. <br>

Другие возможные причины: <br>
** Утеря данных <br>
** Человеческий фактор при внесении данных в базу <br>
** Технические ошибки при внесении данных в базу <br>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* в отличии от средних (арифметическое/геометрическое/гармоническое) медианное значение лучшим образом характеризует выборку в случае, если распределние значений отличается от нормального, т.е. оно менее чувствительно к "вылетающим" низким или высоким значениям. <br>
Замена пропусков на медианное значение позволяет использовать данные, которые иначе использовать было бы нельзя, к тому же такая замена сравнительно слабо влияет на прочие статистические параметры выборки (среднее, стандартное отклонение и пр.).

### Шаг 4: общий вывод.

В данном исследовании были проверены четыре гипотезы о зависимостях надежности заемщиков от их данных:
* Надежность заемщика зависит от количества детей
* Надежность заемщика зависит от семейного положения
* Надежность заемщика зависит от уровня дохода
* Цель получения кредита влияет на возврат его в срок

Получившиеся результаты представлены ниже:

<b>Гипотеза 1. Надежность заемщика зависит от количества детей </b> <br>
Гипотеза <b>подтверждается</b>. <br>
Среди заемщиков без детей доля не вернувших кредит в срок составила около 7.5%, в то время, как аналогичная доля заемщиков с 1 или 2 детьми составила около 9.3%. Количество данных о заемщиках с 3 детьми и более недостаточно для достоверных выводов о данных группах. <br>
На основании всего этого можно сделать вывод, что <b>заемщики без детей существенно более надежны, чем заемщики с детьми. При этом количество детей в случае их наличия не влияет на надежность заемщика.</b>
<br>
<br>
<b>Гипотеза 2. Надежность заемщика зависит от семейного положения</b><br>
Гипотеза <b>подтверждается</b>. <br>
Доля заемщиков с задолжностями, не состоявших в браке (холостые, гражданский брак) составила около 9.5%, в то время, как аналогичная доля для заемщиков, состоящих или ранее состоявших в браке (разведенные, вдовы) составила не более 7.6%. <br>
Соответственно, можно сделать вывод о том, что <b>заемщики, состоящие или ранее состоявшие в браке существенно более надежны, чем заемщики, не состоявшие в браке</b>
<br>
<br>
<b>Гипотеза 3. Надежность заемщика зависит от дохода</b><br>
Для заключения о подтверждаемости гипотезы <b>данных недостаточно</b>. <br>
Ввиду значительного объема пропущенных данных о доходах в исходном датасете (около 10%), а так же малого числа заемщиков из категорий A (доход более 1Е6) и D/E (доход менее 200E3), проверить эту гипотезу не представляется возможным.<br>
Единственный вывод, который можно сделать - заемщики с доходом категории B (от 200E3 до 1E6) более надежны, чем заемщики с доходом категории C (от 50E3 до 200E3).
<br>
<br>
<b>Гипотеза 4. Цель получения кредита влияет на его возврат в срок</b><br>
Гипотеза <b>подтверждается</b>. <br>
Кредиты для операций с автомобилями и кредиты на образование имели сходное число кредитов с задолженностями, около 9.3%, в то время, как кредиты для операций с недвижимостью и для проведения свадеб имели аналогичную долю на уровне 7.3% и 7.9% соответственно.
<br>
<br>
<font size=4><b>Общие выводы и рекомендации</b><br>
* Количество детей заемщика, его семейное положение и цель получения кредита влияют на рискованность кредита и могут быть использованы для построения модели скоринга. <br>
* Требуется дополнительная работа с большим набором данных для установления зависимости между доходом заемщика и рискованностью кредита. На данный момент этот параметр не рекомендуется для использования в модели скоринга.